tests/real/community.rs

```diff
@@ -795,46 +800,183 @@ async fn test_jobs_with_blocking_handlers() {
 }
 
 #[tokio::test(flavor = "multi_thread")]
-async fn test_panic_in_handler() {
+async fn test_panic_and_errors_in_handler() {
     skip_check!();
 
-    let local = "test_panic_in_handler";
+    let job_kind_vs_error_msg = HashMap::from([
+        ("panic_SYNC_handler_str", "panic_SYNC_handler_str"),
+        ("panic_SYNC_handler_String", "panic_SYNC_handler_String"),
+        ("panic_SYNC_handler_int", "job processing panicked"),
+        ("error_from_SYNC_handler", "error_from_SYNC_handler"),
+        ("panic_ASYNC_handler_str", "panic_ASYNC_handler_str"),
+        ("panic_ASYNC_handler_String", "panic_ASYNC_handler_String"),
+        ("panic_ASYNC_handler_int", "job processing panicked"),
+        ("error_from_ASYNC_handler", "error_from_ASYNC_handler"),
+        (
+            "no_handler_registered_for_this_jobtype_initially",
+            "No handler for no_handler_registered_for_this_jobtype_initially",
+        ),
+    ]);
+    let njobs = job_kind_vs_error_msg.keys().len();
+
+    // clean up
+    let local = "test_panic_and_errors_in_handler";
+    let mut c = Client::connect().await.unwrap();
+    let pattern = format!(r#"*\"args\":\[\"{}\"\]*"#, local);
+    c.requeue(
+        MutationTarget::Retries,
+        MutationFilter::builder().pattern(pattern.as_str()).build(),
+    )
+    .await
+    .unwrap();
+    c.queue_remove(&[local]).await.unwrap();
```

Comment:
Why is this necessary?

Reply:
This clean up logic is needed because we are  (well at least I am) re-using the same container when developing locally. Without it, Faktory will re-schedule the failed jobs _after_ the test run, and so the next test run will face this remainders

Reply:
Would you mind leaving a slightly more elaborate comment to that effect?

Reply:
Sure thing